docs/design/2022-06-15-phyical plan explain.md

```diff
@@ -0,0 +1,406 @@
+# TiSpark Design Documents
+
+- Author(s): [Author Name](http://github.com/your-github-id), [Co-Author Name](http://github.com/your-github-id), ...
+- Tracking Issue: https://github.com/pingcap/tispark/issues/2385
+
+## Table of Contents
+
+* [Introduction](#introduction)
+* [Motivation or Background](#motivation-or-background)
+* [Detailed Design](#detailed-design)
+* [Test Design](#test-design)
+
+## Introduction
+
+The physical plan is the physical execution plan in TiSpark. When we use explain in Spark which run with TiSpark, The process of physical plan will be displayed in the terminal if we use `explain` in Spark which run with TiSpark. The current physical plan which is  displayed in the terminal has problems such as outputting multiple filters  which not should show together in the same time,unclear representation of the execution process and obscure push-down conditions. The displayed of physical plan need to be improved. 
+
+## Motivation or Background
+
+### The Process of Scan
+
+#### IndexScan
+
+In TiSpark, An IndexScan requires two scans. One is scanning in index data to get the RowID and the other one is scanning in table data to get the real data. The first scan in code called IndexScan, and the second scan in code called TableScan,but in order to facilitate the distinction,  in the latter part of the text, the first scan is called IndexRangeScan, the second scan is called TableRowIDScan.
```

Comment:
what's the difference between IndexFullScan and IndexRangeScan, and when will IndexFullScan appear in TiDB?

Reply:
> what's the difference between IndexFullScan and IndexRangeScan, and when will IndexFullScan appear in TiDB?

I add explain for it